src/libraries/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -1166,6 +1250,7 @@ public static float Sum(this IQueryable<float> source)
                     CachedReflectionInfo.Sum_NullableSingle_1, source.Expression));
         }
 
+        [DynamicDependency("Sum", typeof(Enumerable))]
```

Comment:
Does this end up keeping all the Sum overloads?  Is there a way to specify it should only keep the one for double?

Reply:
Yes, it will keep all the Sum overloads. It is possible to fully specify the exact method, but it really complicates the signature. And in my opinion, makes this a lot more risky. We do get linker warnings if the method can't be resolved. But if we get the wrong one, we could have a bug.

For example:

```C#
[DynamicDependency("Where`1", typeof(Enumerable))]
```

would now become:

```C#
[DynamicDependency("Where`1(System.Collections.Generic.IEnumerable{``0},System.Func{``0,System.Boolean})", typeof(Enumerable))]
```

Do you think I should go ahead and change all of them to specify the exact signature?

Reply:
Here's the start of what that would look like:

https://github.com/dotnet/runtime/commit/af0cfbd15294b90a28e2bc550c55ecd56ddbba1a

Reply:
If it's risky then it's probably not worth it for this release. Maybe worth revisiting for the next.

Reply:
It does seem more error-prone, but if the signatures would match up such that the unit test you have would find mismatches, then it would be a nice touch. Are there any where the signatures don't match?

But "not worth it for this release" sounds good to me.

Reply:
`Sum` and `Average` are definitely the extreme cases here - there are 10 overloads of each (20 total, but we specify the generic type count in the signature, so the ILLinker will only preserve 10).

For Sum, doing it the way I proposed adds 2.5 KB of IL if you only use a single overload.
For Average, doing it the way I proposed adds 7 KB of IL if you only use a single overload.

> Are there any where the signatures don't match?

Basically all the signatures don't match exactly. `IQueryable` => `IEnumerable`, and `Expression<Func<...>>` => `Func<...>`. So we would need to encode that in the unit test as well.